build-logic/settings/src/main/kotlin/com/squareup/anvil/builds/settings/SettingsPlugin.kt

```diff
@@ -0,0 +1,66 @@
+package com.squareup.anvil.builds.settings
+
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+import org.gradle.api.initialization.dsl.VersionCatalogBuilder
+import org.gradle.api.internal.file.FileOperations
+import java.net.URI
+import javax.inject.Inject
+
+abstract class SettingsPlugin @Inject constructor(
+  private val fileOperations: FileOperations,
+) : Plugin<Settings> {
+
+  override fun apply(target: Settings) {
+    target.dependencyResolutionManagement.versionCatalogs { container ->
+
+      val catalogBuilder = container.maybeCreate("libs")
+
+      if (target.rootProject.name != "anvil") {
```

Comment:
Can you use the `isInMainAnvilBuild()` helper here? Maybe needs to be moved to a more common location first since it's in conventions

Reply:
It looks like this is the same thing, but this `rootProject` is actually of type `ProjectDescriptor`, which isn't even related to `Project`.